packages/internal-test-env/index.ts

```diff
@@ -87,69 +86,57 @@ export interface EnvVariables {
 }
 
 export function setupEnv() {
-  // If we're in CI, the environment is already configured.
-  if (process.env.CI) {
-    return;
-  }
-
   if (envLoaded) {
     return;
   }
+  if (!process.env.CI) {
+    const envPath = join(process.cwd(), "e2e/env/.env.local");
 
-  const envPath = join(process.cwd(), "e2e/env/.env.local");
-
-  // Otherwise load dotenv configuration
-  config({
-    path: envPath,
-  });
+    // Otherwise load dotenv configuration
+    config({
+      path: envPath,
+    });
 
-  if (!process.env.E2E_TEST_ENVIRONMENT) {
-    console.error(
-      `We didn't find the given environment variable E2E_TEST_ENVIRONMENT,
-tried looking in the following directory for \`.env.local \`: 
-${envPath}`
-    );
+    if (!process.env.E2E_TEST_ENVIRONMENT) {
+      console.error(
+        `We didn't find the given environment variable E2E_TEST_ENVIRONMENT,
+  tried looking in the following directory for \`.env.local \`: 
+  ${envPath}`
+      );
+    }
   }
-
-  // Creating feature flag object if there are feature flags
-  Object.keys(process.env)
-    .filter((envVar) => envVar.startsWith("E2E_TEST_FEATURE_"))
-    .forEach((key) => (featuredFlags[key] = process.env[key]));
-  // Marking env as loaded
+  // If we're in CI, the environment is already configured, and we just loaded
+  // it otherwise.
   envLoaded = true;
 }
 
-function getTestingEnvironment(
-  environment: unknown
-): asserts environment is EnvVariables {
-  // Populate your process.env from your .env file of choice
-
-  // TODO: Replace these inline validations and checks with envalid or env-var
-  if (
-    !availableEnvironment.includes(
-      (environment as EnvVariables).E2E_TEST_ENVIRONMENT as AvailableEnvironment
-    )
-  ) {
-    throw new Error(
-      `Unknown environment: [${
-        (environment as EnvVariables).E2E_TEST_ENVIRONMENT
-      }]`
-    );
-  }
-
-  if (typeof (environment as EnvVariables).E2E_TEST_IDP !== "string") {
-    throw new Error("The environment variable E2E_TEST_IDP is undefined.");
-  }
-}
-
 function getBaseTestingEnvironment<T extends LibraryVariables>(libVars?: T
   ): T extends NodeVariables ? TestingEnvironmentNode : TestingEnvironmentBrowser   {
     setupEnv();
-    getTestingEnvironment(process.env);
+    // TODO: Replace these inline validations and checks with envalid or env-var
+    
+    // Load and validate target environment name.
+    const targetEnvName = process.env.E2E_TEST_ENVIRONMENT;
+    if (
+      !availableEnvironment.includes(targetEnvName as AvailableEnvironment)
+    ) {
+      throw new Error(`Unknown environment: [${targetEnvName}]`);
+    }
+
+    // Load and validate target OpenID Provider.
+    const targetOp = process.env.E2E_TEST_IDP;
+    if (typeof targetOp !== "string") {
```

Comment:
Should we URL validate here too?

Reply:
897597c5b8122c01bc5e690e7cbdddf38d31bbcc